lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1140,6 +1139,12 @@ WasmBytecodeGenerator::EmitMemAccess(WasmOp wasmOp, const WasmTypes::WasmType* s
     {
         throw WasmCompilationException(_u("Index expression must be of type I32"));
     }
+
+    if (offset >= m_module->GetMemory()->maxSize)
+    {
+        throw WasmCompilationException(_u("Index is out of bounds"));
```

Comment:
We should have the check but emit an opcode that traps, that way it is a runtime trap not compile error, see my previous comment about this.